### PR TITLE
fix docs for proto library re-generation

### LINKED ISSUE #542: Improve docs on compiling proto files and setting up the dev env
Currently, we don't have clear docs on how to set up dev env for compiling proto files. This issue focuses on making the setting up a developer environment easier.

### PR DESCRIPTION
**Pull request by avinal**
# Changes

- update docs for regenerating proto libraries
- regenerate proto libraries
- fixes #542

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/lgtm

**Review by khrm — COMMENTED** [review]
/lgtm

File: proto/README.md [hunk 1]
```diff
@@ -1,8 +1,64 @@
...
+    ls $HOME/.local/include/google/protobuf
+    any.proto  any_test.proto  api.proto  compiler  descriptor.proto  descriptor_test.proto  empty.proto  struct.proto  struct_test.proto  timestamp.proto  timestamp_test.proto  type.proto  type_test.proto  wrappers.proto  wrappers_test.proto
+    ```
+
+- Install plugins required for Tekton Results proto library generation.
+
+  ```bash
+  go install github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway@latest
+  go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
+  go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
```
Comment (adambkaplan) [inline proto/README.md:41]: Does this imply that `$GOPATH/bin` needs to also be in your `$PATH`?

**Review by adambkaplan — APPROVED** [review]
/approve

File: proto/README.md (hunk 1 above)
Comment (avinal) [reply proto/README.md:41]: Yes, I omitted it since it is a part of the go installation process. I have mentioned it in the first prerequisite, though.

**Pull request merged**